docs/adr/000x-modelling-history.md

```diff
@@ -0,0 +1,250 @@
+- Content Publiser can't show an accurate link or status for the live edition
+  of a document;
+- users cannot edit or remove images on a document once the first
+  edition is published;
+- the TimelineEntry model stores aspects of a documents state, resulting in it
+  needed to be queried outside a timeline context.
+
+And this prevents a number of intended features for Content Publisher:
+
+- comparing different editions of a document;
+- republishing live content if there are any problems (currently a common
+  support task for Whitehall publisher);
+- showing users what changes a user made in a particular edit.
+
+## Decision
+
+This ADR proposes changes to the domain model to resolve the aforementioned
+pain points and provide a means to support the future intended features. It
+defines a number of [core concepts](#core-concepts) that underpin the
+modelling. These provide means to store the individual editions of a document,
+each revision of the content of a document and each status an edition has held.
+As per [ADR-3](0003-initial-domain-modelling.md) it does not consider the
+option of sharing data between translations of a document as there are not
+the appropriate product decisions for this.
+
+A common theme in this modelling is the defining of
+[whether models are mutable](#approach-to-mutabilityimmutability) once
+persisted. Immutability is favoured as a means to preserve a history since
+each change requires storing a new record, implicitly creating a history.
+Mutability is used as a means to simplify interacting with models and preserve
+Rails idioms.
+
+This ADR then considers the impact of history for [timeline](#timeline) and
+[topics](#topics), both areas where the needs for history are less
+clear. Finally, this ADR concludes with an [overview](#overview) of the domain
+model.
+
+### Core Concepts
+
+![Main concepts](000x/main-concepts-diagram.png)
+
+**Document**: A record that represents all versions of a piece of content in a
+particular locale. It has many editions and at any time it will have a current
+edition - shown on Content Publisher index - and potentially a live edition
+which is currently on GOV.UK. The live and current edition can be
+the same. Each iteration of a document's content is represented as a revision,
+thus a document has many revisions. Document is a mutable entity that tracks
+data across all editions (such as first publishing date) and it is expected to
+be a joining point for document-related data that is not associated with a
+particular edition.
+
+**Edition**: A numbered version of a document that has been, or is
+expected to be, published on GOV.UK. It is associated with a revision
+and a status. It is mutable and is a place where any edition level
+database constraints can be placed, such as only one live edition can exist for
+a document. It is possible that two editions of the same document share the
+same revision, this is in the case that they have identical content.
```

Comment:
I struggled how to resolve this concern of yours for a while.

I don't think imply is the correct term as it's explicit rather than implicit that the content is the same. I've tried to capture this in a sentence.